avocado/utils/pci.py

```diff
@@ -0,0 +1,310 @@
+    Gets disks in a pci device.
+
+    :parm device: pci device.
+
+    :return: list of disks in a pci device.
+    """
+    disks_path = "/dev/disk/by-path/"
+    devices_list = []
+    for dev in os.listdir(disks_path):
+        if device in dev:
+            link = os.readlink("%s%s" % (disks_path, dev))
+            devices_list.append(os.path.abspath("%s%s" % (disks_path, link)))
+    return devices_list
+
+
+def get_interfaces_in_device(device):
+    """
+    Gets ethernet interface in a pci device.
+
+    :parm device: pci device.
+
+    :return: list of interfaces in a pci device.
+    """
+    iface_path = "/sys/class/net/"
+    interfaces_list = []
+    for iface in os.listdir(iface_path):
+        if device in os.readlink("%s%s" % (iface_path, iface)):
+            interfaces_list.append(iface)
+    return interfaces_list
+
+
+def get_pci_fun_list(device):
+    """
+    Gets list of functions in the given pci device.
+    Example: in device 0000:03:00, functions are 0000:03:00.0 and 0000:03:00.1
+
+    :parm device: pci device.
+
+    :return: list of functions in a pci device.
+    """
+    return list(dev for dev in get_pci_devices() if device in dev)
+
+
+def get_slot_from_sysfs(device):
+    """
+    Gets the pci slot of given device.
+    Specific for ppc64 processor
+
+    :parm device: pci device.
+
+    :return: slot of pci device from sysfs.
+    """
+    if not os.path.isdir('/sys/bus/pci/devices/%s' % device):
+        return
+    cmd = "cat /sys/bus/pci/devices/%s/devspec" % device
+    devspec = process.system_output(cmd, shell=True).strip('\n')
+    cmd = "cat /proc/device-tree/%s/ibm,loc-code" % devspec
+    slot = process.system_output(cmd, shell=True).strip('\0\n')
+    return re.match(r'((\w+)[\.])+(\w+)-(\w*\d+)-(\w*\d+)|Slot(\d+)',
+                    slot).group()
+
+
+def get_pci_id_from_sysfs(device):
+    """
+    Gets the pci id from sysfs of given device.
+
+    :parm device: pci device.
+
+    :return: pci id of a pci device from sysfs.
+    """
+    path = "/sys/bus/pci/devices/%s" % device
+    if os.path.isdir(path):
+        path = "%s/%%s" % path
+        return ":".join(["%04x" % int(open(path % param).read(), 16)
+                         for param in ['vendor', 'device', 'subsystem_vendor',
+                                       'subsystem_device']])
+
+
+def get_slot_list():
+    """
+    Gets list of pci slots in the system.
+
+    :return: list of slots in the system.
+    """
+    return list(set(get_slot_from_sysfs(dev) for dev in get_pci_devices()))
+
+
+def get_specific_pci_id(device, part):
+    """
+    Gets specific pci id of given divice.
+
+    :parm device: pci device.
+    :parm part: part of pci id.
+
+    :return: specific pci id of a pci device.
+    """
+    cmd = "lspci -Dnvmm -s %s" % device
+    output = process.system_output(cmd, shell=True, ignore_status=True)
+    if output:
+        for line in output.splitlines():
+            if part == line.split()[0].strip(':'):
```

Comment:
I agree in most cases this would be false, so it makes sense to use the split twice, but then the first split can be simpler:

```
if part == line.split(':', 1):
    return line.split()[-1]
```